designs/ami-minimum-age.md

```diff
@@ -0,0 +1,70 @@
+# Ami Minimum Age
+
+This document describes a design for a new feature to allow users to specify a minimum age for AMIs.
+
+(a @jonathan-innis's idea, see [this comment](https://github.com/aws/karpenter-provider-aws/issues/5382#issuecomment-1868068193))
+
+## Background
+
+Karpenter allows users to specify TTL for nodes, which configures the maximum age of nodes in the cluster. 
```

Comment:
I think you're referring to expiration here, which I don't think really has relevance here. In the case of an AMISelector Term that doesn't use AMI ID, we look periodically at if the AMI has changed every 5 minutes, and roll that out. This minimumAge would be a way to modify our discovery logic to change what that resolved AMI would be.

Reply:
my understanding is that a new AMI doesn't trigger a drift, so nodes running on the old AMI won't be recycled. 
So setting an expiration makes sure that the entire cluster is bounced to the new AMI, over a certain period of time.
Is this correct, or when a new AMI is detected, the nodes running the old AMI are marked as drifted?

Reply:
found this in the documentation: https://karpenter.sh/docs/concepts/disruption/#special-cases-on-drift but not sure if it's related also to configurations where no `amiSelectorTerms` is set?